tests/validation_suite/automation_framework/work_order_submit/work_order_submit_params.py

```diff
@@ -0,0 +1,643 @@
+        out_data = self.get_out_data()
+
+        self.hash_2 = ""
+        if in_data is not None:
+            self.hash_2 = self._compute_hash_string(in_data)
+
+        self.hash_3 = ""
+        if out_data is not None:
+            self.hash_3 = self._compute_hash_string(out_data)
+
+        final_string = self.hash_1 + self.hash_2 + self.hash_3
+        self.final_hash = crypto.compute_message_hash(
+                          bytes(final_string, 'UTF-8'))
+
+        self.encrypted_request_hash = self.byte_array_to_hex_str(
+                                      self.encrypt_data(
+                                      self.final_hash, self.session_key,
+                                      self.session_iv))
+
+        self.params_obj["encryptedRequestHash"] = self.encrypted_request_hash
+
+    def _compute_hash_string(self, data):
+        final_hash_str = ""
+        hash_string = ""
+        for data_item in data :
+            data = "".encode('UTF-8')
+            datahash = "".encode('UTF-8')
+            e_key = "".encode('UTF-8')
+            iv = "".encode('UTF-8')
+            if 'dataHash' in data_item:
+                datahash = data_item['dataHash'].encode('UTF-8')
+            if 'data' in data_item:
+                data = data_item['data'].encode('UTF-8')
+            if 'encryptedDataEncryptionKey' in data_item:
+                e_key = \
+                data_item['encryptedDataEncryptionKey'].encode('UTF-8')
+            if 'iv' in data_item:
+                iv = data_item['iv'].encode('UTF-8')
+            hash_string =  datahash + data + e_key + iv
+            complete_bytes = bytes(hash_string)
+            hash = crypto.compute_message_hash(complete_bytes)
+            final_hash_str = final_hash_str + crypto.byte_array_to_base64(hash)
+        return final_hash_str
+
+    def _compute_requester_signature(self):
+        self.public_key =  self.private_key.GetPublicKey().Serialize()
+        signature_result =  self.private_key.SignMessage(self.final_hash)
+        self.requester_signature  =  crypto.byte_array_to_base64(
+                                     signature_result)
+        self.params_obj["requesterSignature"] = self.requester_signature
+        self.params_obj["verifyingKey"] = self.public_key
+
+    def byte_array_to_hex_str(self, in_byte_array):
+        '''
+        Converts tuple of bytes to hex string
+        '''
+        logger.debug("Input Byte Array: %s", in_byte_array)
+        hex_str = ''.join(format(i, '02x') for i in in_byte_array)
+        logger.debug("Output Byte Array to str: %s", hex_str)
+        return hex_str
+
+    def strip_begin_end_key(self, key) :
+        """
+        Strips off newline chars, BEGIN PUBLIC KEY and END PUBLIC KEY.
+        """
+        return key.replace("\n", "").replace(
+               "-----BEGIN PUBLIC KEY-----", "").replace(
+               "-----END PUBLIC KEY-----", "")
+
+    def strip_rsa_begin_end_key(self, key) :
+        """
```

Comment:
This is same as utility in source code. Can you directly take the utility function instead of duplicating this.